src/main/java/org/dependencytrack/tasks/vulnerabilitypolicy/blobstorage/NginxStorageHandler.java

```diff
@@ -0,0 +1,51 @@
+package org.dependencytrack.tasks.vulnerabilitypolicy.blobstorage;
+
+import alpine.common.logging.Logger;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.dependencytrack.tasks.vulnerabilitypolicy.AbstractVulnerabilityPolicyFetcher;
+import org.dependencytrack.util.VulnerabilityPolicyUtil;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.zip.ZipInputStream;
+
+public class NginxStorageHandler extends AbstractVulnerabilityPolicyFetcher {
+
+    private static final Logger LOGGER = Logger.getLogger(NginxStorageHandler.class);
+
+    public NginxStorageHandler() {
+        super.setUsername();
+        super.setPassword();
+    }
+
+    @Override
+    public boolean verifyDownloadNeeded() throws IOException {
+        try (CloseableHttpResponse response = performHeadRequest()) {
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                Header[] httpHeaders = response.getAllHeaders();
+                Optional<Header> headerOptional = Arrays.stream(httpHeaders).filter(header -> header.getName().equalsIgnoreCase("ETag")).findFirst();
+                if (headerOptional.isPresent()) {
+                    return VulnerabilityPolicyUtil.matchWithHashConfigProperty(headerOptional.get().getValue());
+                } else {
+                    LOGGER.warn("Was not able to find Etag header in the nginx request. Will proceed assuming that file needs to be downloaded");
+                    return true;
+                }
+            } else {
+                throw new IOException("Unable to get response from resource bundle endpoint for policy");
```

Comment:
Can we provide more information here? The exception is thrown when the response status is not `200`, so what was it *actually*?